runtime/cudaq/builder/kernel_builder.cpp

```diff
@@ -630,7 +630,9 @@ ExecutionEngine *jitCode(ImplicitLocOpBuilder &builder, ExecutionEngine *jit,
   pm.addPass(cudaq::opt::createExpandMeasurementsPass());
   pm.addPass(createCanonicalizerPass());
   pm.addPass(cudaq::opt::createApplyOpSpecializationPass());
-  pm.addPass(cudaq::opt::createLoopUnrollPass());
+  optPM.addPass(cudaq::opt::createClassicalMemToReg());
```

Comment:
Question - does the LowerToQIR work for Quake in the value semantic model? I'm guessing it does.

Reply:
I don't know honestly. I haven't tried that.

Reply:
Well the tests passed here...

Reply:
`bin/cudaq-quake ../scratch/test.cpp | bin/cudaq-opt --canonicalize --cse  --memtoreg | bin/cudaq-translate --convert-to=qir` on something like 
```cpp
auto super = []() __qpu__ {
    cudaq::qreg q(1);
    h(q[0]);
    mz(q[0]);
  };
```
leads to 
```mlir
; ModuleID = 'LLVMDialectModule'
source_filename = "LLVMDialectModule"
target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-unknown-linux-gnu"

%Array = type opaque

declare void @__quantum__rt__qubit_release_array(%Array*) local_unnamed_addr

declare %Array* @__quantum__rt__qubit_allocate_array(i64) local_unnamed_addr

define void @"__nvqpp__mlirgen__Z4mainE3$_0"() local_unnamed_addr {
  %1 = tail call %Array* @__quantum__rt__qubit_allocate_array(i64 1)
  tail call void @__quantum__rt__qubit_release_array(%Array* %1)
  ret void
}

!llvm.module.flags = !{!0}

!0 = !{i32 2, !"Debug Info Version", i32 3}
```

Reply:
So it may be that we'll have to add back a Reg2Mem pass on the kernel_builder.

Reply:
I guess this works because its just a classical mem-to-reg...